### ISSUE TITLE
Question: Strange "Indent expected and not found compared"

### ISSUE THREAD

**Issue by mrshu** [alanmcruickshank/sqlfluff#180]
When running the `master` on the following

```sql
WITH ttt AS (
    SELECT
        t.val AS value
    FROM test AS t
)

SELECT * FROM ttt
```

the result I got was the following:

```
L:   5 | P:   1 | L003 | Indent expected and not found compared to line #1 
```

but the query that would be acceptable was the following:

```sql
WITH ttt AS (
    SELECT
        t.val AS value
    FROM test AS t
    )

SELECT * FROM ttt
```

I would personally prefer the first one, as it looks a bit more pleasant to read. Similarly, I honestly do not like the following `after.sql` file (https://github.com/alanmcruickshank/sqlfluff/blob/a54fbe71a016575b2ee7eb588ef899f5464a4664/test/fixtures/linter/autofix/ansi/001_L016_long_line_1/after.sql) as it makes it very difficult to notice where does the `WITH` statement end.

Would you mind sharing a bit of background for this? I'd be eager to understand it better.

Thanks a lot!

**Comment by alanmcruickshank** [comment]
Good challenge.

The reason for the current implementation is for the case where there are multiple CTEs in the `WITH` statement. For example: 

```
WITH ttt AS (
        SELECT
            t.val AS value
        FROM test AS t
    ),
    qqq AS (
        SELECT * FROM foo
    )
    
SELECT * FROM ttt JOIN qqq USING (value)
```

The closing bracket after `WITH` in your example query isn't closing the `WITH` statement, it's just closing the definition of `ttt`. One option that might solve this would be to remove the idea of an indent *before* ttt, which would mean the idea formatting for the example above would be:

```
WITH ttt AS (
    SELECT
        t.val AS value
    FROM test AS t
),
qqq AS (
    SELECT * FROM foo
)
    
SELECT * FROM ttt JOIN qqq USING (value)
```

I'm not sure I like `qqq` being on the same level as `WITH`, but it wouldn't be too hard to do and it's a workable solution. @mrshu, which do you prefer in the two CTE example?

**Comment by alanmcruickshank** [comment]
Interestingly that would also be consistent with the fishtown analytics guidlines https://github.com/fishtown-analytics/corp/blob/master/dbt_coding_conventions.md#ctes

So I'd be happy to go with that if it works for you?

**Comment by mrshu** [comment]
@alanmcruickshank despite sharing many of the reservations you've outlined, I'd prefer the second one you mentioned as that's what I've been used to for quite some time.

Thanks a ton for considering it!

**Comment by alanmcruickshank** [comment]
@mrshu - followup question...

at the moment, based on how I've implemented the indentation rules, both:
```
WITH ttt AS (
    SELECT
        t.val AS value
    FROM test AS t
)

SELECT * FROM ttt
```
and
```
WITH ttt AS (
    SELECT
        t.val AS value
    FROM test AS t)

SELECT * FROM ttt
```

... are both valid. Now I quite like your argument for having that closing bracket on the same line as `WITH`, even if technically it's ok in both positions. I'm considering implementing a linting rule which says that although this layout is fine just from a bracketing perspective, there's a specific linting rule for `WITH` clauses, which says that the closing bracket in this kind of case should *always* be on the same level of indentation as the `WITH` keyword. i.e. the first example above is correct, and the second example would raise a linting violation.

Is that too draconian, or is it actually a helpful linting rule to help people write better sql?

**Comment by alanmcruickshank** [comment]
@mrshu - I've put together a PR to make the changes you've suggested - I think it should end up with results closer to what you initially suggested.

Let me know what you think of the potential for a linting rule in my previous comment. If you think it's a good idea, then I'll add it to the same PR before merging.

**Comment by mrshu** [comment]
@alanmcruickshank I personally do think it's a very good idea -- it seems that [Mozilla's SQL style guide](https://docs.telemetry.mozilla.org/concepts/sql_style.html#parentheses) would agree as well (they even use the same example as you do).

Thanks again for considering it!

**Comment by alanmcruickshank** [comment]
Changes are merged as part of #182, including a linting rule for the indentation of the closing bracket. I'm going to close this issue. Thanks for your input @mrshu !
